src/components/dialog/dialog.js

```diff
@@ -1,107 +1,293 @@
+import {keyCodes, defaultClassNames, NATIVELY_FOCUSABLE_ELEMENTS} from '../../constants';
+
+import createEl from '../../utils/createEl';
+import defer from '../../utils/defer';
+import qa from '../../utils/qa';
+
+
 /**
  * @function UIDialog
  * @version 0.0.2
+ * @desc Main UIDialog function. Creates instances of the dialog based on
+ * parameters passed in.
+ * @param {object} settings
  */
-
-import {keyCodes, defaultClassNames} from '../../constants';
-
-
 const UIDialog = ({
         dialog = '.js-dialog',
         openBtn = '.js-dialog-btn',
         closeBtn = '.js-dialog-close-btn',
         isModal = false,
+        isAlert = false,
+        readyClass = defaultClassNames.IS_READY,
+        activeClass = defaultClassNames.IS_ACTIVE,
         showBackdrop = true
     } = {}) => {
 
-    let DOM,
-        state = {};
+    // Stores all the dom nodes for the module
+    let DOM = {
+        dialogs: qa(dialog)
+    };
+
+    // Keeps track of current state
+    let state = {
+        currentOpenButton: null,
+        currentDialog: null,
+        focusableElements: null //  elements within modal
+    };
 
-    // Create the backdrop
-    const backdrop = document.createElement('div');
 
-    backdrop.classList.add(defaultClassNames.DIALOG_BACKDROP);
+    /**
+     * @function bindBackdropEvents
+     * @desc Adds event listener for clicking on backdrop
+     */
+    function bindBackdropEvents() {
+        DOM.backdrop.addEventListener('click', handleBackdropClick);
+    }
 
-    init();
 
     /**
-     * @function init
-     * @desc Initialises the dialog
+     * @function unbindBackdropEvents
+     * @desc Removes event listener for clicking on backdrop
      */
-    function init() {
+    function unbindBackdropEvents() {
+        DOM.backdrop.removeEventListener('click', handleBackdropClick);
+    }
 
-        // Save all DOM queries for future use
-        DOM = {
-            'page':     document.querySelectorAll('body')[0],
-            'dialog':   document.querySelectorAll(dialog)[0],
-            'openBtn':  document.querySelectorAll(openBtn)[0],
-            'closeBtn': document.querySelectorAll(closeBtn)[0]
-        };
 
-        // Check if the dialog exists, return if not
-        if (DOM.dialog === undefined) {
-            return false;
+    /**
+     * @function handleBackdropClick
+     * @desc If backdrop has been clicked, dismiss dialog
+     * @param {Event} e
+     */
+    function handleBackdropClick(e) {
+        if (e.target === DOM.backdrop) {
```

Comment:
No, because we don't attach the event listener if it's a modal in the first place.  Around line 160 we have this:

```
if (!isModal && DOM.backdrop) {
    defer(bindBackdropEvents);
}
```